synapse/lib/snap.py

```diff
@@ -411,22 +424,47 @@ def addNodes(self, nodedefs):
         Returns:
             (list): A list of xact messages.
         '''
+        with self.bulkload():
 
-        for (formname, formvalu), forminfo in nodedefs:
+            for (formname, formvalu), forminfo in nodedefs:
 
-            props = forminfo.get('props')
-            node = self.addNode(formname, formvalu, props=props)
+                props = forminfo.get('props')
 
-            tags = forminfo.get('tags')
-            if tags is not None:
-                for tag, asof in tags.items():
-                    node.addTag(tag)
+                # remove any universal created props...
+                if props is not None:
+                    props.pop('.created', None)
+
+                node = self.addNode(formname, formvalu, props=props)
+
+                tags = forminfo.get('tags')
+                if tags is not None:
+                    for tag, asof in tags.items():
+                        node.addTag(tag)
 
     def stor(self, sops):
         if not self.write:
-            raise s_exc.ReadOnlyXact()
+            raise s_exc.ReadOnlySnap()
+
+        if self.bulk:
+            self.bulksops.extend(sops)
+            return
+
+        self.xact.stor(sops)
+
+    @contextlib.contextmanager
+    def bulkload(self):
 
-        self.layr._xactRunStors(self.xact, sops)
+        #mine = not self.bulk
+        #if mine:
+            #self.bulk = True
+            #self.bulksops = []
+
+        yield
```

Comment:
yield None?

Reply:
this feels incomplete

Reply:
Sure.  The explicit syntax is probably better.